relayer/src/supervisor.rs

```diff
@@ -119,11 +127,76 @@ impl Supervisor {
         })
     }
 
+    fn create_workers(&mut self) -> Result<(), BoxError> {
+        let req = QueryChannelsRequest {
+            pagination: ibc_proto::cosmos::base::query::pagination::all(),
+        };
+
+        // For each opened channel spawn:
+        // - the path worker for AtoB (used to clear packets) and
+        // - the client worker for the A side client of the channel's connection (used for refresh)
+        // TODO - we should move to one supervisor, operating on a set of chains:
+        // `for chain_a in self.chains.clone().iter() {`, lookup chain b in registry based
+        // on the client chain_id, etc.
+        for chains in [self.chains.clone(), self.chains.clone().swap()].iter() {
```

Comment:
Since Rust 1.51 we can use [`std::array::IntoIter`](https://doc.rust-lang.org/stable/std/array/struct.IntoIter.html) to iterate over an array by value:

```suggestion
        use std::array;
        let chains_pairs = [self.chains.clone(), self.chains.clone().swap()];
        for chains in array::IntoIter(chains_pair) {
```